beaconChain/lib/stateTransition/processBlock.ts

```diff
@@ -1,9 +1,391 @@
+      [
+        blsAggregatePubkeys(custodyBit0Participants.map((i) => state.validatorRegistry[i].pubkey)),
+        blsAggregatePubkeys(custodyBit1Participants.map((i) => state.validatorRegistry[i].pubkey)),
+      ],
+      [
+        treeHash({ data: attestation.data, custodyBit: false } as AttestationDataAndCustodyBit),
+        treeHash({ data: attestation.data, custodyBit: true } as AttestationDataAndCustodyBit)
+      ],
+      attestation.aggregateSignature,
+      getDomain(state.fork, slotToEpoch(attestation.data.slot), Domain.ATTESTATION)
+    );
+    if (!custodyBitsVerified) {
+      throw new Error();
+    }
+    // Remove the following conditional in Phase 1
+    if (attestation.data.shardBlockRoot.equals(ZERO_HASH)) {
+      throw new Error();
+    }
+    state.latestAttestations.push({
+      data: attestation.data,
+      aggregationBitfield: attestation.aggregationBitfield,
+      custodyBitfield: attestation.custodyBitfield,
+      inclusionSlot: state.slot,
+    } as PendingAttestation);
+  }
+
+  // Deposits
+  if (!(block.body.deposits.length <= MAX_DEPOSITS)) {
+    throw new Error();
+  }
+  // TODO: add logic to ensure that deposits from 1.0 chain are processed in order
+  for (const deposit of block.body.deposits) {
+    const serializedDepositData = serialize(deposit.depositData, DepositData);
+    if (!deposit.index.eq(state.depositIndex)) {
+      throw new Error();
+    }
+    if (!verifyMerkleBranch(hash(serializedDepositData), deposit.branch, DEPOSIT_CONTRACT_TREE_DEPTH, deposit.index.toNumber(), state.latestEth1Data.depositRoot)) {
+      throw new Error();
+    }
+    processDeposit(
+      state,
+      deposit.depositData.depositInput.pubkey,
+      deposit.depositData.amount,
+      deposit.depositData.depositInput.proofOfPossession,
+      deposit.depositData.depositInput.withdrawalCredentials
+    );
+    state.depositIndex = state.depositIndex.addn(1);
+  }
+
+  // Voluntary Exits
+  if (!(block.body.voluntaryExits.length <= MAX_VOLUNTARY_EXITS)) {
+    throw new Error();
+  }
+  for (const exit of block.body.voluntaryExits) {
+    const validator = state.validatorRegistry[exit.validatorIndex.toNumber()];
+    if (!(validator.exitEpoch.gt(getEntryExitEffectEpoch(currentEpoch)))) {
+      throw new Error();
+    }
+    if (!currentEpoch.gte(exit.epoch)) {
+      throw new Error();
+    }
+    const exitMessage = treeHash({
+      epoch: exit.epoch,
+      validatorIndex: exit.validatorIndex,
+      signature: EMPTY_SIGNATURE
+    } as VoluntaryExit);
+    const exitMessageVerified = blsVerify(
+      validator.pubkey,
+      exitMessage,
+      exit.signature,
+      getDomain(state.fork, exit.epoch, Domain.EXIT)
+    );
+    if (!exitMessageVerified) {
+      throw new Error();
+    }
+    initiateValidatorExit(state, exit.validatorIndex);
+  }
+
+  // Transfers
+  // Note: Transfers are a temporary functionality for phases 0 and 1, to be removed in phase 2.
```

Comment:
for reasons such as this, as well as testing and readability, does it not make sense to break these out into a set of functions?

Reply:
Yeah! There are pretty clear chunks that can/should be broken out into their own functions/files. One for each processing step and one for each transaction type.